apps/api/src/app/workflows-v2/usecases/generate-preview/generate-preview.usecase.ts

```diff
@@ -1,68 +1,156 @@
 import {
   ChannelTypeEnum,
   GeneratePreviewRequestDto,
   GeneratePreviewResponseDto,
   JobStatusEnum,
+  JSONSchemaDto,
   PreviewPayload,
   StepDataDto,
-  UserSessionData,
+  WorkflowOriginEnum,
 } from '@novu/shared';
+import {
+  GetWorkflowByIdsCommand,
+  GetWorkflowByIdsUseCase,
+  WorkflowInternalResponseDto,
+} from '@novu/application-generic';
 import { PreviewStep, PreviewStepCommand } from '../../../bridge/usecases/preview-step';
 import { FrameworkPreviousStepsOutputState } from '../../../bridge/usecases/preview-step/preview-step.command';
-import { StepMissingControlsException } from '../../exceptions/step-not-found-exception';
-import { PrepareAndValidateContentUsecase, ValidatedContentResponse } from '../validate-content';
 import { BuildStepDataUsecase } from '../build-step-data';
 import { GeneratePreviewCommand } from './generate-preview.command';
+import { extractTemplateVars } from '../../util/template-variables/extract-template-variables';
+import { pathsToObject } from '../../util/path-to-object';
 
 @Injectable()
 export class GeneratePreviewUsecase {
   constructor(
     private legacyPreviewStepUseCase: PreviewStep,
-    private prepareAndValidateContentUsecase: PrepareAndValidateContentUsecase,
-    private buildStepDataUsecase: BuildStepDataUsecase
+    private buildStepDataUsecase: BuildStepDataUsecase,
+    private getWorkflowByIdsUseCase: GetWorkflowByIdsUseCase
   ) {}
 
   async execute(command: GeneratePreviewCommand): Promise<GeneratePreviewResponseDto> {
     const dto = command.generatePreviewRequestDto;
     const stepData = await this.getStepData(command);
+    const workflow = await this.findWorkflow(command);
 
-    const validatedContent: ValidatedContentResponse = await this.getValidatedContent(dto, stepData, command.user);
+    const variablesExample = this.buildVariablesExample(workflow, stepData, dto);
     const executeOutput = await this.executePreviewUsecase(
       command,
       stepData,
-      validatedContent.finalPayload,
-      validatedContent.finalControlValues
+      variablesExample,
+      dto.controlValues || {}
     );
 
     return {
-      issues: validatedContent.issues,
       result: {
         preview: executeOutput.outputs as any,
         type: stepData.type as unknown as ChannelTypeEnum,
       },
-      previewPayloadExample: validatedContent.finalPayload,
-    };
+      previewPayloadExample: variablesExample,
+    } as GeneratePreviewResponseDto;
   }
 
-  private async getValidatedContent(dto: GeneratePreviewRequestDto, stepData: StepDataDto, user: UserSessionData) {
-    if (!stepData.controls?.dataSchema) {
-      throw new StepMissingControlsException(stepData.stepId, stepData);
+  private buildVariablesExample(
+    workflow: WorkflowInternalResponseDto,
+    stepData: StepDataDto,
+    dto: GeneratePreviewRequestDto
+  ) {
+    const variablesExample = this.generateVariablesExample(dto, stepData);
+
+    if (workflow.origin === WorkflowOriginEnum.EXTERNAL) {
```

Comment:
Not all code-first workflows have payload ;)
```suggestion
    if (workflow.payloadSchema) {
```

Reply:
this one is connected to 
https://github.com/novuhq/novu/pull/7137/files#r1859187874
and be resolved there.